importing all our librarys:

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
import re
import warnings
warnings.filterwarnings("ignore")

importing our data:

In [2]:
df = pd.read_csv("All London 2012 athletes cleaning.csv")
df.reset_index(drop=True, inplace=True) #reset the index, while row are delete

In [3]:
df.head()

Name                     Country  Age  Height, cm  Weight Sex  \
0         Lamusi A  People's Republic of China   23       170.0    60.0   M   
1       A G Kruger    United States of America   33       193.0   125.0   M   
2   Jamale Aarrass                      France   30       187.0    76.0   M   
3  Maria Abakumova          Russian Federation   26       178.0    85.0   F   
4   Emanuele Abate                       Italy   27       190.0    80.0   M   

  Date of birth          Place of birth  Gold  Silver  Bronze  Total  \
0      6/2/1989        NEIMONGGOL (CHN)     0       0       0      0   
1     2/18/1979           Sheldon (USA)     0       0       0      0   
2    11/15/1981            BEZONS (FRA)     0       0       0      0   
3     1/15/1986  STAVROPOL REGION (RUS)     0       0       0      0   
4      7/8/1985            Genova (ITA)     0       0       0      0   

       Sport                  Event  
0       Judo            Men's -60kg  
1  Athletics     Men's Hammer Throw  
2  Athletics            Men's 1500m  
3  Athletics  Women's Javelin Throw  
4  Athletics     Men's 110m Hurdles

In [4]:
df.describe(include='all')

Name                   Country          Age   Height, cm  \
count               6495                      6495  6495.000000  6495.000000   
unique              6495                       196          NaN          NaN   
top     Abdouraim Haroun  United States of America          NaN          NaN   
freq                   1                       349          NaN          NaN   
mean                 NaN                       NaN    26.391070   175.810470   
std                  NaN                       NaN     5.756301    10.765394   
min                  NaN                       NaN    13.000000   132.000000   
25%                  NaN                       NaN    23.000000   168.000000   
50%                  NaN                       NaN    26.000000   175.000000   
75%                  NaN                       NaN    29.000000   183.000000   
max                  NaN                       NaN    71.000000   218.000000   

             Weight   Sex Date of birth Place of birth         Gold  \
count   6495.000000  6495          6495           6495  6495.000000   
unique          NaN     2          4169           3624          NaN   
top             NaN     M     2/22/1988    Seoul (KOR)          NaN   
freq            NaN  3539             7             46          NaN   
mean      71.216012   NaN           NaN            NaN     0.017552   
std       16.521416   NaN           NaN            NaN     0.141486   
min       36.000000   NaN           NaN            NaN     0.000000   
25%       60.000000   NaN           NaN            NaN     0.000000   
50%       68.000000   NaN           NaN            NaN     0.000000   
75%       80.000000   NaN           NaN            NaN     0.000000   
max      218.000000   NaN           NaN            NaN     2.000000   

             Silver       Bronze        Total      Sport           Event  
count   6495.000000  6495.000000  6495.000000       6495            6495  
unique          NaN          NaN          NaN         34             673  
top             NaN          NaN          NaN  Athletics  Women's Hockey  
freq            NaN          NaN          NaN       1898             210  
mean       0.017398     0.019554     0.054503        NaN             NaN  
std        0.135388     0.140677     0.265773        NaN             NaN  
min        0.000000     0.000000     0.000000        NaN             NaN  
25%        0.000000     0.000000     0.000000        NaN             NaN  
50%        0.000000     0.000000     0.000000        NaN             NaN  
75%        0.000000     0.000000     0.000000        NaN             NaN  
max        2.000000     2.000000     5.000000        NaN             NaN

**we have a problem in column 'Place of birth', the uniqe values are 3624 values.
so we will fix it by split the country and city, using only the country the athlete came from:**

In [5]:
pob= df['Place of birth']
pob

0             NEIMONGGOL (CHN)
1                Sheldon (USA)
2                 BEZONS (FRA)
3       STAVROPOL REGION (RUS)
4                 Genova (ITA)
                 ...          
6490             Gulbene (LAT)
6491                    SHIRAZ
6492         VLADIKAVKAZ (RUS)
6493               Kranj (SLO)
6494             Liberec (CZE)
Name: Place of birth, Length: 6495, dtype: object

In [6]:
for i in range(0,len(pob)):
    newVal= re.split("[()]", pob[i])
    try:
        pob[i]= newVal[1]
    except:
        pob[i]= np.nan
df['Place of birth'] = pob

In [7]:
df.describe(include='all')

Name                   Country          Age   Height, cm  \
count               6495                      6495  6495.000000  6495.000000   
unique              6495                       196          NaN          NaN   
top     Abdouraim Haroun  United States of America          NaN          NaN   
freq                   1                       349          NaN          NaN   
mean                 NaN                       NaN    26.391070   175.810470   
std                  NaN                       NaN     5.756301    10.765394   
min                  NaN                       NaN    13.000000   132.000000   
25%                  NaN                       NaN    23.000000   168.000000   
50%                  NaN                       NaN    26.000000   175.000000   
75%                  NaN                       NaN    29.000000   183.000000   
max                  NaN                       NaN    71.000000   218.000000   

             Weight   Sex Date of birth Place of birth         Gold  \
count   6495.000000  6495          6495           5812  6495.000000   
unique          NaN     2          4169            194          NaN   
top             NaN     M     2/22/1988            USA          NaN   
freq            NaN  3539             7            367          NaN   
mean      71.216012   NaN           NaN            NaN     0.017552   
std       16.521416   NaN           NaN            NaN     0.141486   
min       36.000000   NaN           NaN            NaN     0.000000   
25%       60.000000   NaN           NaN            NaN     0.000000   
50%       68.000000   NaN           NaN            NaN     0.000000   
75%       80.000000   NaN           NaN            NaN     0.000000   
max      218.000000   NaN           NaN            NaN     2.000000   

             Silver       Bronze        Total      Sport           Event  
count   6495.000000  6495.000000  6495.000000       6495            6495  
unique          NaN          NaN          NaN         34             673  
top             NaN          NaN          NaN  Athletics  Women's Hockey  
freq            NaN          NaN          NaN       1898             210  
mean       0.017398     0.019554     0.054503        NaN             NaN  
std        0.135388     0.140677     0.265773        NaN             NaN  
min        0.000000     0.000000     0.000000        NaN             NaN  
25%        0.000000     0.000000     0.000000        NaN             NaN  
50%        0.000000     0.000000     0.000000        NaN             NaN  
75%        0.000000     0.000000     0.000000        NaN             NaN  
max        2.000000     2.000000     5.000000        NaN             NaN

**#now we can see the unique values of column 'Place of birth' is 19, a number we can work with.**

**#we need to drop the columns that dont give us important information for our research**

In [8]:
df.drop(['Name', 'Country', 'Date of birth', 'Event'], axis=1, inplace=True)
df.dropna(axis=0, inplace= True)

In [9]:
df.reset_index(drop=True, inplace=True) #we will reset again after delete
df.head()

Age  Height, cm  Weight Sex Place of birth  Gold  Silver  Bronze  Total  \
0   23       170.0    60.0   M            CHN     0       0       0      0   
1   33       193.0   125.0   M            USA     0       0       0      0   
2   30       187.0    76.0   M            FRA     0       0       0      0   
3   26       178.0    85.0   F            RUS     0       0       0      0   
4   27       190.0    80.0   M            ITA     0       0       0      0   

       Sport  
0       Judo  
1  Athletics  
2  Athletics  
3  Athletics  
4  Athletics

# data treatment- letters to numeric

**# check which category need our treatment and see what kind o possible values we have**

In [10]:
print('Age:', sorted(list(df['Age'].unique())))
print('Height, cm:', sorted(list(df['Height, cm'].unique())))
print('Weight:', sorted(list(df['Weight'].unique())))
print('Sex:', sorted(list(df['Sex'].unique())))
print('Place of birth:', sorted(list(df['Place of birth'].unique())))
print('Gold:', sorted(list(df['Gold'].unique())))
print('Silver:', sorted(list(df['Silver'].unique())))
print('Bronze:', sorted(list(df['Bronze'].unique())))
print('Total:', sorted(list(df['Total'].unique())))
print('Sport:', sorted(list(df['Sport'].unique())))

Age: [13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 56, 57, 58, 65, 71]
Height, cm: [132.0, 135.0, 140.0, 145.0, 147.0, 148.0, 149.0, 150.0, 151.0, 152.0, 153.0, 154.0, 155.0, 156.0, 157.0, 158.0, 159.0, 160.0, 161.0, 162.0, 163.0, 164.0, 165.0, 166.0, 167.0, 168.0, 169.0, 170.0, 171.0, 172.0, 173.0, 174.0, 175.0, 176.0, 177.0, 178.0, 179.0, 180.0, 181.0, 182.0, 183.0, 184.0, 185.0, 186.0, 187.0, 188.0, 189.0, 190.0, 191.0, 192.0, 193.0, 194.0, 195.0, 196.0, 197.0, 198.0, 199.0, 200.0, 201.0, 202.0, 203.0, 204.0, 205.0, 206.0, 207.0, 208.0, 209.0, 210.0, 211.0, 212.0, 218.0]
Weight: [36.0, 38.0, 39.0, 40.0, 41.0, 42.0, 43.0, 44.0, 45.0, 46.0, 47.0, 48.0, 49.0, 50.0, 51.0, 52.0, 53.0, 54.0, 55.0, 56.0, 57.0, 58.0, 59.0, 60.0, 61.0, 62.0, 63.0, 64.0, 65.0, 66.0, 67.0, 68.0, 69.0, 70.0, 71.0, 72.0, 73.0, 74.0, 75.0, 76.0, 77.0, 78.0, 79.0, 80.0, 81.0, 82.0, 83.0, 84.0,

we can see, that we have 3 categories that need our treatment:
### 1. change the values of sex to binary (men/female)
### 2. change the sport to numeric values (1-34)
### 3. change the 'place of birth column to numeric values (1-194) 

In [11]:
df= pd.get_dummies(df, columns=['Sex'], prefix=['Sex'])

In [15]:
numericSport = dict(zip(range(1,37),df.Sport.unique()))
numericSport = dict(zip(numericSport.values(), numericSport.keys()))
df['Sport'] = df['Sport'].map(numericSport).fillna(df['Sport'])

In [16]:
country_labels = df["Place of birth"].astype('category').cat.categories.to_list()
rep_map={x:y for x,y in zip(country_labels,list(range(0,len(country_labels)+1)))}
df["Place of birth"].replace(rep_map, inplace=True)

In [17]:
df.head()

Age  Height, cm  Weight  Place of birth  Gold  Silver  Bronze  Total  \
0   23       170.0    60.0              37     0       0       0      0   
1   33       193.0   125.0             186     0       0       0      0   
2   30       187.0    76.0              62     0       0       0      0   
3   26       178.0    85.0             151     0       0       0      0   
4   27       190.0    80.0              89     0       0       0      0   

   Sport  Sex_F  Sex_M  
0      1      0      1  
1      2      0      1  
2      2      0      1  
3      2      1      0  
4      2      0      1

### all data are fixed and numberd !

In [19]:
df.to_csv("All London 2012 athletes numeric only.csv", index=False)